library/alloc/src/slice/tests.rs

```diff
@@ -0,0 +1,359 @@
+use crate::borrow::ToOwned;
```

Comment:
This whole file is the subset of `alloc/test/slice.rs` that uses non-determinism. There are clearly some things in it than need cleaning up, but I'd rather do that at a future time.

Reply:
Making the test helper public would have worked too. Since it's gated on cfg(test) it would only be available here but not to crate users.

Reply:
Hm? `cfg(test)` isn't enabled for integration tests, only unit tests.

Reply:
Huh? I thought I had seen something like that before. Maybe it was through a perma-unstable feature instead. But nevermind, if moving the code is the simplest option, so be it.

Reply:
Yeah I'd definitely prefer copypaste to exposing a type from an external crate in std. Even as perma-unstable. One option to avoid the duplication would be `include!`ing a file, though. But I don't know if that's better. If the reviewer would prefer it I don't mind, but IDK where the file would go (and also it needs to be slightly different for std vs everything else, since `std::blah` doesn't work in std)